### PR TITLE
add retry mechanism to ConnectTracker and modify Listen backlog to 128

### PR DESCRIPTION
**Pull request by daiyl0320**
通过测试，详见issue描述https://github.com/dmlc/xgboost/issues/544

### CODE REVIEW

**Comment by tqchen** [comment]
One question about the backlog. 
- The retry did helps in this setting
- Seems we want to increase the backlog of tracker instead of the worker here?

File: subtree/rabit/src/allreduce_base.cc [hunk 1]
```diff
@@ -15,6 +15,11 @@
 
 namespace rabit {
 namespace engine {
+
+namespace {
+const int kConnectRetryTimes = 5;
```
Comment (tqchen) [inline subtree/rabit/src/allreduce_base.cc]: Consider make this configurable, like other parameters, via DMLC_WORKER_CONNECT_RETRY

**Comment by tqchen** [comment]
Thanks for the contribution! Can you send the PR to https://github.com/dmlc/rabit instead? This will be pulled back as subtree of rabit

**Comment by daiyl0320** [comment]
OK，我修改一下。关于backlog，你说的没错，需要在rabit_tracker.py中修改listen的backlog

**Pull request closed without merging**